src/t_stream.c

```diff
@@ -210,6 +211,16 @@ robj *streamDup(robj *o) {
             new_nack->delivery_count = nack->delivery_count;
             new_nack->cgroup_ref_node = streamLinkCGroupToEntry(new_s, new_cg, ri_cg_pel.key);
             raxInsert(new_cg->pel, ri_cg_pel.key, sizeof(streamID), new_nack, NULL);
+
+            streamID id;
+            streamDecodeID(ri_cg_pel.key, &id);
+
+            uint64_t keyBuf[3];
+            pelTimeKey timeKey;
+            timeKey.delivery_time = new_nack->delivery_time;
+            timeKey.id = id;
+            encodePelTimeKey(keyBuf, &timeKey);
```

Comment:
is it possible to avoid the need to create pelTimeKey each time when encodePelTimeKey?
```suggestion
            uint64_t keyBuf[3];
            encodePelTimeKey(keyBuf, new_nack->delivery_time, &id);
```

Reply:
No need of this. I've created two new helper functions, raxInsertPelByTime and raxRemovePelByTime, which encapsulate the encoding and insert/removal operations for the pel_by_time rax tree. This reduces code duplication and simplifies the implementation.